### PR TITLE
[WNMGDS-2020] updates for hint text from DSM

### PR DESCRIPTION
**Pull request by line47**
## Summary

- Update hint text guidance from HCgov DSM

## How to test

1. run `yarn start` and visit form label page and see that hint text has guidance has been updated.

## Checklist

- [x] Prefixed the PR title with the [Jira ticket number](https://jira.cms.gov/projects/WNMGDS/) as `[WNMGDS-####] Title`

### If this is a change to code:

- [x] Created or updated unit tests to cover the logic added
- [x] If necessary, updated unit-test snapshots (`yarn test:unit:update`) and browser-test snapshots (`yarn test:browser:update`)

### If this is a change to documentation:

- [x] Checked for spelling and grammatical errors

### CODE REVIEW

File: packages/docs/content/components/form-label.mdx [hunk 1]
```diff
@@ -34,37 +34,49 @@ The following Sass variables can be overridden to customize Form components:
...
-
-#### Labels
-
-- Each form field should have a `<label>`. Never use a field's placeholder attribute as the primary way to label the field.
-
-#### Legends
+### Form legend
 
 - Use a single legend for fieldset (this is required). One example of a common use of fieldset and legend is a question with radio button options for answers. The question text and radio buttons are wrapped in a fieldset, with the question itself being inside the legend tag.
+- Avoid links in legends when possible. Links inside legends are not read by some screen readers, most notably Internet Explorer 11, paired with JAWS. If you need a pattern like this, put the link outside of the legend element, but keep all other useful hint text in the legend.
```
Comment (pwolfert) [inline packages/docs/content/components/form-label.mdx:40]: Is this valid now that we no longer support IE 11? @SparrowSpit, do you know of any other software combos that would make this necessary? Or is it possibly a bad pattern in general?

File: packages/docs/content/components/form-label.mdx [hunk 2]
```diff
@@ -34,37 +34,49 @@ The following Sass variables can be overridden to customize Form components:
...
 - Don't use a field's placeholder attribute as the primary way to label the field.
 
-#### Legends
+#### Content
 
-- Avoid links in legends when possible. Links inside legends are not read by some screen readers, most notably Internet Explorer 11, paired with JAWS. If you need a pattern like this, put the link outside of the legend element, but keep all other useful hint text in the legend.
+- Describe what the user needs to do when the field is selected
+- Use a phrase or complete sentence
+- Start with an action verb and pair it with a noun to help it be clearer. Example: Enter your age.
+- Use the least amount of words possible (i.e. don't use "please")
```
Comment (pwolfert) [inline packages/docs/content/components/form-label.mdx:58]:
```suggestion
- Use the fewest words possible (i.e. don't use "please")
```

File: packages/docs/content/components/form-label.mdx [hunk 3]
```diff
@@ -34,37 +34,49 @@ The following Sass variables can be overridden to customize Form components:
...
 
-#### Hint text
+### Hint text
 
 - Place hint text within the field's `<label>` element.
-- Apply the `.ds-c-field__hint` class to hint text.
+- Place hint text where appropriate depending on what content it's helping to explain
 - Use hint text for supporting contextual help, which will always be shown.
-- Hint text should sit above a form field and below the label text.
+- Hint text should sit between a from label and the input.
```
Comment (pwolfert) [inline packages/docs/content/components/form-label.mdx:68]:
```suggestion
- Hint text should sit between a form label and the input.
```

File: packages/docs/content/components/form-label.mdx [hunk 4]
```diff
@@ -34,37 +34,49 @@ The following Sass variables can be overridden to customize Form components:
...
+- Use lower visual style (i.e., lighter gray than the form label or the input option) so it won't compete for users' attention with the field label. 
+- Use hint text to indicate which inputs are optional when there are more required fields than optional.
+- Hint text can also appear after an input box when the text is being used to show the value someone should be entering in the field (e.g., income amount input box displays text after the input box to highlight the amount should be "monthly income".)
+
+#### Content
+
+- Use when appropriate to support a form label (Goes below the form label in a lighter weight font)
+- Describe how to best answer the question/form label
+- Use the least amount of words possible (i.e. don't use "please")
```
Comment (pwolfert) [inline packages/docs/content/components/form-label.mdx:77]:
```suggestion
- Use the fewest words possible (i.e. don't use "please")
```

File: packages/docs/content/components/form-label.mdx (hunk 2 above)
Comment (zarahzachz) [reply packages/docs/content/components/form-label.mdx:58]: Does this bullet point contradict the one 2 points before (use a phrase or complete sentence)?

Do we care if folks to use numerically fewer words, or are we asking them to use direct/active language when writing labels?

Also, the "i.e." here should probably be "e.g." as "please" is being used as an example.

File: packages/docs/content/components/form-label.mdx (hunk 1 above)
Comment (SparrowSpit) [reply packages/docs/content/components/form-label.mdx:40]: [It looks like this is a known issue](https://github.com/FreedomScientific/VFO-standards-support/issues/402) and I also tested and experienced the issue described in that convo. 

> You tab into a button within the element. The legend button is read.
You tab into a button underneath the contained within the fieldset. The fieldset button is read.
You shift-tab back to the button within the element.
The previous button is read back to you.

Comment (pwolfert) [reply packages/docs/content/components/form-label.mdx:40]: Okay, so it's also a problem for Chrome, Firefox, and Edge according to that ticket. Thanks, @SparrowSpit! Maybe we should update it to say 
```suggestion
- Avoid links in legends when possible. Links inside legends are not read by some screen readers, namely JAWS. If you need a pattern like this, put the link outside of the legend element, but keep all other useful hint text in the legend.
```

**Review by pwolfert — APPROVED** [review]

**Pull request merged**